Source/DafnyStandardLibraries/src/Std/TargetSpecific/Duration.dfy

```diff
@@ -0,0 +1,405 @@
+include "DateTimeConstant.dfy"
+include "../Collections/SeqExt.dfy"
+
+module Std.Duration {
+  import opened DateTimeConstant
+  import opened Strings
+  import opened Collections.SeqExt
+  import opened Collections.Seq
+  import opened BoundedInts
+  import opened Arithmetic.Power
+
+  datatype Duration = Duration(
+    seconds: uint16,
```

Comment:
I was imprecise with a previous comment and led you astray, and I apologize. I was suggesting that since `millis` by definition is at most 1000, you could use a `uint16` for it. But I didn't mean to imply that for `seconds`, which has no natural upper bound. I would use a `uint64` for `seconds` since that's large enough for basically any practical use, and still cheap and efficient in most compiled contexts.